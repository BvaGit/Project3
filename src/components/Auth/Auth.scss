@import '../../utils/variables.scss';
$width: 600px;
.gen {
    height: 100vh;
    @include flex(row, center, center);
    &__container {
        width: 80%;
        margin: 0 auto;
    }
    &__holder {
        margin: 0 auto;
        width: 800px;
        height: 100%;
    }
    &__info {
        width: $width;
        height: fit-content;
    }
    &__main {
        width: $width;
        height: fit-content;
        background: $background;
        border-radius: 12px;
    }
    &__buttons {
        @include flex(row, space-around, center);
        height: 20%;
    }
    &__link {
        display: block;
        width: 100%;
        text-align: right;
        color: $link-color;
    }
}
.container {
    padding: 0 $padding_lg;
}
