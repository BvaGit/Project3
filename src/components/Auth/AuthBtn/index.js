import {connect} from "react-redux";

import AuthBtn from "./AuthBtn";
import { getIsValidAuthFields } from "../../../store/auth/selectors";
import {authUserRequest} from "../../../store/auth/asyncActions";

const mapStateToProps = state => ({
    isError: getIsValidAuthFields(state),
})

const mapDispatchToProps = dispatch => ({
    handleAuth: () => dispatch(authUserRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AuthBtn);
