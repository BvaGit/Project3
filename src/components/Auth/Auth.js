import React from "react";
import { NavLink } from "react-router-dom";

import Logo from "../Logo";
import Title from "../Title";
import AuthInput from "./AuthInput";
import AuthBtn from "./AuthBtn";

import "./Auth.scss";

const Auth = () => {
  return (
    <div className="gen__holder">
      <div className="gen">
        <div className="gen__info">
          <Logo src="/assets/images/logo.png" alt="logo" />
          <Title textBig="SIGN IN" textSmall="YOUR ACCOUNT" />
        </div>
        <div className="gen__main">
          <div className="gen__container">
            <AuthInput
              id="login"
              type="text"
              label="Login"
              title="Enter your login"
              placeholder="Enter your login"
            />
            <AuthInput
              id="password"
              type="password"
              label="Password"
              title="Enter your password"
              placeholder="Enter your password"
            />
            <div className="gen__buttons">
              <AuthBtn />
            </div>
            <p className="gen__link">Create new account</p>
            <NavLink to="/registration" className="gen__link">
              Sign up
            </NavLink>
          </div>
        </div>
      </div>
    </div>
  ); 
};

export default Auth;
