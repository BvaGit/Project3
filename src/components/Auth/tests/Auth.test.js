import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from "react-intl";
import { Router } from 'react-router-dom'


import Auth from '../Auth';
import configureStore from 'redux-mock-store';


const mockStore = configureStore();
const store = mockStore({
    auth : {
        fields: {
          login: "",
          password: "",
        },
        errors: {
          login: "",
          password: "",
        }
      },
    rooms: {
        sendingMessage: false,
        messages: {
          0: [
            {
              text: "Welcome!",
            },
          ],
        },
        socket: null,
        rooms: {},
        activeRoom: "",
      },
      user:{
        user: { id: null, token: null },
      },
      users: {
        users: [],
      },
      registration: {
        fields:{
          login: '',
          password: '',
          confirm_password:""
        },
        errors:{
          login:"",
          password:"",
          confirm_password:""
        }
      },
      location: {value: 'en'}
    });


describe('Auth', () => {
    it("Should match snapshot", () => {
        const component = shallow(<Auth />)
        expect(component).toMatchSnapshot()
    })
    // it("Should have MainPage", () => {
    //     const component = mountBiggerSmart(<Auth />, store)
    //     expect(component.find('div')).toHaveLength(1)
    //   })
})

const mountSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return mount(core);
  };

  const mountBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}><Router>{component}</Router></Provider>
        : component;
    return mount(<IntlProvider >{core}</IntlProvider>);
};