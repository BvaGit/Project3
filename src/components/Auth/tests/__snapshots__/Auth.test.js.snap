// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Should match snapshot 1`] = `ShallowWrapper {}`;
